### PR TITLE
GIDS vettec backend changes

### PR DESCRIPTION
**Pull request by hughes-dustin**
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/18894

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/18895

Description:
Adding the vettec provider indicator to the Institutions table and adding a vet_tec_provider filter to the api institutions endpoint. 

Testing:
QA and dev tested.

### CODE REVIEW

**Review by devin-mccurdy — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Both stories pass QA

**Comment by hughes-dustin** [comment]
@department-of-veterans-affairs/backend-review-group Could I get a code review on this at your earliest convenience. Thank you!

File: spec/controllers/v0/institutions_controller_spec.rb [hunk 1]
```diff
@@ -123,6 +123,12 @@
       expect(JSON.parse(response.body)['data'].count).to eq(1)
     end
 
+    it 'filters by vet_tec_provider schools' do
+      create(:institution, :vet_tec_provider, vet_tec_provider: true)
```
Comment (jholton) [inline spec/controllers/v0/institutions_controller_spec.rb]: Suggestion: create another record where `vet_tec_provider: false` to show that they are not returned:
```suggestion
      create(:institution, :vet_tec_provider, vet_tec_provider: true)
      create(:institution, :vet_tec_provider)
```

File: app/models/institution_builder.rb [hunk 2]
```diff
@@ -19,6 +19,15 @@ def self.columns_for_update(klass)
     klass::COLS_USED_IN_INSTITUTION.map(&:to_s).map { |col| %("#{col}" = #{table_name}.#{col}) }.join(', ')
   end
 
+  def self.add_vet_tec_provider(version_number)
```
Comment (jholton) [inline app/models/institution_builder.rb]: Can we add a spec to `spec/models/institution_builder_spec` to test this method?

**Review by jholton — CHANGES_REQUESTED** [review]

File: app/models/institution_builder.rb (hunk 2 above)
Comment (hughes-dustin) [reply app/models/institution_builder.rb]: I went ahead and added a test in spec/models/institution_builder_spec where we are checking to see that vet_tec_provider  is being set to true for an institution when it's weams record has a facility code with a 'V' as it's second character.

File: spec/controllers/v0/institutions_controller_spec.rb (hunk 1 above)
Comment (hughes-dustin) [reply spec/controllers/v0/institutions_controller_spec.rb]: Added the other record to the test.

File: spec/controllers/v0/institutions_controller_spec.rb [hunk 3]
```diff
@@ -123,6 +123,13 @@
...
```
Comment (jholton) [inline spec/controllers/v0/institutions_controller_spec.rb]: Looks like `vet_tec_provider` is specified as `true` in the factory for `:vet_tec_provider`, so we don't need to specify it here:
```suggestion
      create(:institution, :vet_tec_provider)
```

Comment (hughes-dustin) [reply spec/controllers/v0/institutions_controller_spec.rb]: Added the change.

**Review by jholton — APPROVED** [review]
LGTM

**Pull request merged**